nslsii/EPS_devices/valves_and_shutters.py

```diff
@@ -0,0 +1,363 @@
+        The string value to be passed to ``RE(mv(device, val))`` to move to
+        state 1
+    state2_str : str, optional
+        The string value to be passed to ``RE(mv(device, val))`` to move to
+        state 2
+    state1_pv_uid : str, optional
+        The unique part of the EPICS PV for the state 1 command(see note below)
+    state2_pv_uid : str, optional
+        The unique part of the EPICS PV for the state 2 command(see note below)
+    num_retries : int, optional
+        The number of attempts at changing the state prior to raising an
+        error, the default is 1.
+    retry_sleep_time : float, optional
+        This is the time in seconds to wait between retries on changing the
+        state, the default is 0.5 s.
+    stop_str : str or None, optional
+        The value of ``state1_str`` or ``state2_str`` to be called in the
+        ``stop`` method. The default, if None is entered, is ``state2_str``.
+    **kwargs : dict, optional
+        The kwargs passed to the ``ophyd.Device`` ``__init__`` method
+
+    ..note ::
+
+        The PV associated with moving to 'state 1' is derived from the
+        ``prefix`` and the ``state1_pv_uid`` using the following format:
+        ``'{prefix}Cmd:{state1_pv_uid}-Cmd'``
+   '''
+
+    def __init__(self, *, prefix, name,
+                 state1_val='Open', state2_val='Closed',
+                 state1_str='open', state2_str='close',
+                 state1_pv_uid='Opn', state2_pv_uid='Cls',
+                 num_retries=1, retry_sleep_time=0.5,
+                 stop_str=None, **kwargs):
+
+        self._state1_pv_uid = state1_pv_uid
+        self._state2_pv_uid = state2_pv_uid
+
+        super().__init__(prefix=prefix, name=name, **kwargs)
+
+        self._set_st = None
+        self.read_attrs = ['status']
+        self._state1_val = state1_val
+        self._state2_val = state2_val
+        self._state1_str = state1_str
+        self._state2_str = state2_str
+        self._num_retries = num_retries
+        self._retry_sleep_time = retry_sleep_time
+        if stop_str == state1_str:
+            self._stop_str = self._state1_str
+            self._resume_str = self._state2_str
+        elif stop_str == state2_str or stop_str is None:
+            self._stop_str = self._state2_str
+            self._resume_str = self._state1_str
+        else:
+            raise nslsiiValves_and_ShuttersValueError(
+                'the kwarg ``stop_str`` in the EpicsTwoStateDevice class'
+                '``__init__`` method needs to be None or needs to match one of'
+                'the kwargs ``state1_str`` or ``state2_str``')
+
+        self._cmd_map = {self._state1_str: self._state1_cmd,
+                         self._state2_str: self._state2_cmd}
+        self._target_map = {self._state1_str: self._state1_val,
+                            self._state2_str: self._state2_val}
+
+        self._time_fmtstr = '%Y-%m-%d %H:%M:%S'
+
+    _state1_cmd = FmtCpt(EpicsSignal,
```

Comment:
can you put a default kind on everything?  I suspect that `status` should be 'normal', `enabled_status` should be 'configuration' and everything else should be omitted.

the `_fail_to_state` may also want to be normal?

Reply:
done